File: SnakeWaterGun.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int snakeWaterGun(char you, char comp)
{
    if (you == comp)
    {
        return 0;
    }

    if (you == 's' && comp == 'w')
    {
        return 1;
    }
    else if (you == 'w' && comp == 's')
    {
        return -1;
    }
    else if (you == 'g' && comp == 'w')
    {
        return -1;
    }
    else if (you == 'w' && comp == 'g')
    {
        return 1;
    }
    else if (you == 'g' && comp == 's')
    {
        return 1;
    }
    else if (you == 's' && comp == 'g')
    {
        return -1;
    }
}
int main()
{
    char you, comp;
    int number;
    srand(time(0)); //Number Generator
    number = rand() % 100 + 1;
    if (number <= 33)
    {
        comp = 's';
    }
    else if (number > 33 && number <= 66)
    {
        comp = 'w';
    }
    else
    {
        comp = 'g';
    }

    printf("Enter s for snake, w for water and g for gun: \n");
    scanf("%c", &you);
    int result = snakeWaterGun(you, comp);
    printf("You choose %c and computer choose %c\n", you, comp);

    if (result == 0)
    {
        printf("Match Draw\n");
    }
    else if (result == 1)
    {
        printf("You Win\n");
    }
    else
    {
        printf("You Loose\n");
    }
    return 0;
}

File: Python/Games/SnakeWaterGun.py
#  Snake Water Gun or Rock Paper Scissor.
import random

def gameWin(comp,you):
    # If two value are equal declare a tie.
    if comp == you:
        return None
    # Check for all possibilities when computer chooses s.
    elif comp=='s':
        if you == 'w':
            return False
        else:
            return True
    # Check for all possibilities when computer chooses w.
    elif comp=='w':
        if you == 'g':
            return False
        else:
            return True
    # Check for all possibilities when computer chooses g.
    elif comp=='g':
        if you == 's':
            return False
        else:
            return True
    

randNo = random.randint(1,3)
if randNo==1:
    comp = 's'
elif randNo==2:
    comp = 'w'
else:
    comp = 'g'

you = input("Player's Turn: Snake(s) , water(w) or gun(g) ? : ")

a=gameWin(comp,you)

print(f"\nYou choose {you}\n")
print(f"computer choose {comp}\n")

if a==None:
    print("Game Draw....")
elif a==False:
    print("Computer Win !!") 
elif a==True:
    print("You win !!")
